qemu/tests/live_snapshot_basic.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+from virttest import utils_test
+from virttest import error_context
+from avocado.core import exceptions
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+class LiveSnapshotBase(object):
+
+    """
+    Provide basic functions for live snapshot test cases.
+    """
+
+    def __init__(self, params, env):
+        """
+        Init the default values for live snapshot object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.params = params
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+        self.snapshot_file = params.get("snapshot_file")
+        self.snapshot_format = params.get("snapshot_format", "qcow2")
+        self.snapshot_mode = params.get("snapshot_mode", "absolute-paths")
+        self.node_name = params.get("node_name")
+        self.snapshot_node_name = params.get("snapshot_node_name")
+
+    def create_image(self):
+        """
+        Create a image.
+        """
+        image_name = self.params.get("image_name")
+        self.params['image_name_snapshot'] = image_name + "-snap"
+        image_name_snapshot = self.params['image_name_snapshot']
+        snapshot_params = self.params.object_params("snapshot")
+        base_dir = self.params.get("images_base_dir", data_dir.get_data_dir())
+
+        image_io = QemuImg(snapshot_params, base_dir, image_name)
+        image_name, _ = image_io.create(snapshot_params)
+        return image_name
+
+    def get_base_image(self):
```

Comment:
better to make target image device configurable, so that we can easy to switch target image from system disk to data disk.